core/tests/webdriverio_desktop/diagnosticTestPage.js

```diff
@@ -16,60 +16,137 @@
+    await topicEditorPage.submitTopicThumbnail('../data/test2_svg.svg', true);
+    await topicEditorPage.updateMetaTagContent('topic meta tag');
+    await topicEditorPage.updatePageTitleFragment('topic page title');
+    await topicEditorPage.saveTopic('Added thumbnail.');
+    var url = await browser.getUrl();
+    var topicId = url.split('/')[4].slice(0, -1);
+    await general.closeCurrentTabAndSwitchTo(handle);
+    await adminPage.editConfigProperty(
+      'The details for each classroom page.',
+      'List',
+      async function(elem) {
+        elem = await elem.editItem(0, 'Dictionary');
+        elem = await elem.editEntry(4, 'List');
+        elem = await elem.addItem('Unicode');
+        await action.setValue(
+          'Topic ID', elem, topicId, false);
+      });
+    await classroomPage.get('math');
+    // Even if the topic is unpublished, an unclickable tile is shown
+    // currently.
+    await classroomPage.expectNumberOfTopicsToBe(1);
+    await topicsAndSkillsDashboardPage.get();
+    (
+      await
+      topicsAndSkillsDashboardPage.createSkillWithDescriptionAndExplanation(
+        'Skill 1', 'Concept card explanation', false));
+    await skillEditorPage.addRubricExplanationForDifficulty(
+      'Easy', 'Second explanation for easy difficulty.');
+    await skillEditorPage.saveOrPublishSkill('Edited rubrics');
+    // A minimum of three questions are required for skill to get assigned in
+    // a topic’s diagnostic test.
+    await workflow.createQuestion();
+    await workflow.createQuestion();
+    await workflow.createQuestion();
+
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.navigateToSkillsTab();
+    await topicsAndSkillsDashboardPage.assignSkillToTopic(
+      'Skill 1', 'Topic 1');
+    await topicsAndSkillsDashboardPage.get();
+    await topicsAndSkillsDashboardPage.navigateToTopicWithIndex(0);
+
+    await topicEditorPage.addDiagnosticTestSkill('Skill 1');
+
+    await topicEditorPage.addSubtopic(
+      'Subtopic 1', 'subtopic-one', '../data/test2_svg.svg',
+      'Subtopic content');
+    await topicEditorPage.saveTopic('Added subtopic.');
+
+    await topicEditorPage.navigateToTopicEditorTab();
+    await topicEditorPage.navigateToReassignModal();
+
+    await topicEditorPage.dragSkillToSubtopic('Skill 1', 0);
+    await topicEditorPage.saveRearrangedSkills();
+    await topicEditorPage.saveTopic('Added skill to subtopic.');
+
+    await topicEditorPage.publishTopic();
+    await classroomPage.get('math');
+    await classroomPage.expectNumberOfTopicsToBe(1);
+
+    await browser.url('/classroom-admin/');
+    await waitFor.pageToFullyLoad();
+    await diagnosticTestPage.createNewClassroomConfig('Math', 'math');
   });
 
-  beforeEach(async function() {
-    await users.login('creator@classroomPage.com');
+  afterAll(async function() {
+    await users.logout();
   });
 
   it('should be able to start diagnostic test', async function() {
```

Comment:
Shouldn't you have at least one test that reaches the "completion" modal and verify that recommendations are actually shown to the user? Otherwise your tests here have no expectations.